httpcore/_async/http_proxy.py

```diff
@@ -140,47 +140,56 @@ def __init__(
         connection = await self._get_connection_from_pool(origin)
 
         if connection is None:
-            connection = AsyncHTTPConnection(
-                origin=origin, http2=False, ssl_context=self._ssl_context,
+            proxy_connection = AsyncHTTPConnection(
+                origin=self.proxy_origin, http2=False, ssl_context=self._ssl_context,
             )
             async with self._thread_lock:
                 self._connections.setdefault(origin, set())
-                self._connections[origin].add(connection)
+                self._connections[origin].add(proxy_connection)
 
             # Establish the connection by issuing a CONNECT request...
 
             # CONNECT www.example.org:80 HTTP/1.1
             # [proxy-headers]
             target = b"%b:%d" % (url[1], url[2])
             connect_url = self.proxy_origin + (target,)
-            connect_headers = self.proxy_headers
-            proxy_response = await connection.request(
+            connect_headers = self.proxy_headers or [(b"host", self.proxy_origin[1])]
```

Comment:
The host header is required by h11 but not typically passed as part of the proxy headers.

Not sure if we want to merge the headers here or leave that to the caller to sort out.